src/client/components/filter-menu/time-filter-menu/fixed-time-tab.tsx

```diff
@@ -118,10 +119,11 @@ export class FixedTimeTab extends React.Component<FixedTimeTabProps, FixedTimeTa
   }
 
   isFilterDifferent(): boolean {
-    const { essence: { filter, timeShift } } = this.props;
+    const { essence: { filter, timeShift }, dimension } = this.props;
     const newTimeShift = this.constructTimeShift();
-    const newFilter = this.constructFixedFilter();
-    return !filter.equals(newFilter) || !timeShift.equals(newTimeShift);
+    const oldClause = filter.getClauseForDimension(dimension);
```

Comment:
We have a guarantee that oldClause exists, but this guarantee is not expressed in type system. Something to improve sometimes.